cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/CustomActionProgramWorkflowRunner.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright © 2014-2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.internal.app.runtime.workflow;
+
+import co.cask.cdap.api.workflow.WorkflowAction;
+import co.cask.cdap.api.workflow.WorkflowActionSpecification;
+import co.cask.cdap.api.workflow.WorkflowSpecification;
+import co.cask.cdap.api.workflow.WorkflowToken;
+import co.cask.cdap.app.program.Program;
+import co.cask.cdap.app.runtime.ProgramOptions;
+import co.cask.cdap.internal.app.runtime.ProgramRunnerFactory;
+
+/**
+ * A {@link ProgramWorkflowRunner} that creates {@link Runnable} for executing custom action from Workflow.
+ */
+final class CustomActionProgramWorkflowRunner extends AbstractProgramWorkflowRunner {
```

Comment:
Instead of extending AbstractProgramWorkflowRunner, this class can simply implement the ProgramWorkflowRunner and create method of the implementation can return the runnable.